dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/ContextFilter.java

```diff
@@ -125,7 +126,11 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         }
 
         if (invocation instanceof RpcInvocation) {
-            ((RpcInvocation) invocation).setInvoker(invoker);
+            RpcInvocation rpcInvocation = (RpcInvocation) invocation;
+            rpcInvocation.setInvoker(invoker);
+            //write all attachments back;
+            rpcInvocation.getObjectAttachments().putAll(context.getObjectAttachments());
+
```

Comment:
It would be better to fill in the attachment when creating rpcInvocation, so as to avoid the performance loss caused by multiple coverage. If it is to pass information from the rest filter, you can extend a field in RpcServiceContext, obtain it in AbstractProxyInvoker and use it when creating a new invocation.